plugins/orchestrator/src/components/next/WorkflowRunListContent.tsx

```diff
@@ -0,0 +1,158 @@
+
+const DASHES = '---';
+interface Row {
+  id: string;
+  name: string;
+  workflow: string;
+  status: string;
+  started: string;
+  duration: string;
+  component: string;
+}
+
+const DetailPanel = ({ rowData }: { rowData: Row }) => {
+  return (
+    <div>
+      TODO - render timeline component based on {JSON.stringify(rowData)}
+    </div>
+  );
+};
+
+export const WorkflowRunListContent = () => {
+  const orchestratorApi = useApi(orchestratorApiRef);
+  const workflowInstanceLink = useRouteRef(workflowInstanceRouteRef);
+  const [workflow, setWorkflow] = useState<string>();
+  const [status, onChangeStatus] = useState<string>();
+  const [isExpanded, setIsExpanded] = React.useState(true);
+
+  const { loading, error, value } = useAsync(async () => {
+    const instances = await orchestratorApi.getInstances();
+    const clonedData: Row[] = instances
+      .filter(
+        instance =>
+          !instance.parentProcessInstanceId /* TODO: verify that this is truely filtering top-level instances */,
+      )
+      .map(instance => {
+        const start = moment(instance.start?.toString());
+        const end = moment(instance.end?.toString());
+        const duration = moment.duration(start.diff(end));
+        const name = instance.processName || instance.processId; /* TODO */
+
+        const row: Row = {
+          id: instance.id,
+          name,
+          workflow: instance.processName || instance.processId,
+          started: start.format('MMMM DD, YYYY'),
+          duration: duration.humanize(),
+          status: instance.state,
+          component: instance.source || DASHES /* TODO: is that correct? */,
+        };
+
+        return row;
+      });
+
+    return clonedData;
+  }, [orchestratorApi]);
+
+  const columns = React.useMemo(
+    (): TableColumn<Row>[] => [
+      {
+        title: 'Name',
+        render: data => (
+          <Link to={workflowInstanceLink({ instanceId: data.id })}>
+            {data.name}
+          </Link>
+        ),
+      },
+      { title: 'Workflow', field: 'workflow' },
+      {
+        title: 'Status',
+        render: data => <ProcessInstanceStatus status={data.status} />,
+      },
+      { title: 'Started', field: 'started' },
+      { title: 'Duration', field: 'duration' },
+      { title: 'Component', field: 'component' },
+    ],
+    [workflowInstanceLink],
+  );
+
+  const workflows: SelectItem[] = React.useMemo(
+    () =>
+      [
+        // deduplicate
+        ...new Set(value?.map(row => row.workflow)),
+      ]
+        .sort((a, b) => a.localeCompare(b))
+        .map(
+          // make the resulting SelectItem
+          name => ({ label: name, value: name }),
+        ),
+    [value],
+  );
+
+  const filteredData = React.useMemo(
+    () =>
+      (value || []).filter(
+        (row: Row) =>
+          (!workflow || row.workflow === workflow) &&
+          (!status || row.status === status),
+      ),
+    [status, value, workflow],
```

Comment:
```suggestion
    [status, value, workflow],
```
Getting better, however it will still re-render because of `value` (it is always a new array of Rows)

Reply:
The `useAsync` will call the fetch just once

Reply:
Not exactly. It will call fetch once as long as its dependencies array remain the same. In this case it depends on the `orchestratorApi` object which is returned by the `useApi` hook, that in turn gets called every time the component renders. Unless it is guaranteed somewhere it will always return the same object (instead of a new instance), it will cause a re-render